src/test/java/echoserver/FakeSocket.java

```diff
@@ -0,0 +1,31 @@
+package echoserver;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.util.ArrayList;
+
+class FakeSocket extends Socket {
+     ArrayList<StringAsBytes> inputs = new ArrayList<StringAsBytes>();
+     ByteArrayOutputStream os = new ByteArrayOutputStream();
```

Comment:
`os` makes me think *Operating System`. Most generally, short variable names like this make it harder to understand the code. `output` or `outputStream` aren't that much longer and make it very clear what the variable holds.